server/src/db/migrations/20220119232016_store-front.ts

```diff
@@ -0,0 +1,123 @@
+import { Knex } from "knex";
+
+
+export async function up(knex: Knex): Promise<void> {
+
+    knex.schema.hasTable('InventoryItem').then(function (exists) {
+        if (!exists) {
+            return knex.schema.createTable('InventoryItem', function (t) {
+                t.increments('id').primary();
+                t.text('image');
+                t.text('name');
+                t.text('unit');
+                t.text('pluralUnit');
+                t.integer('count');
+                t.float('pricePerUnit');
+            });
+        }
+    });
+
+    // if we know every label we can use an enum instead of this table
+    knex.schema.hasTable('Label').then(function (exists) {
+        if (!exists) {
+            return knex.schema.createTable('Label', function (t) {
+                t.increments('id').primary();
+                t.text('label');
+            });
+        }
+    });
+
+    knex.schema.hasTable('InventoryItemLabel').then(function (exists) {
+        if (!exists) {
+            return knex.schema.createTable('InventoryItemLabel', function (t) {
+                t.increments('id').primary();
+                t.integer('item')
+                    .references('InventoryItem.id')
+                    .onDelete('CASCADE');
+                t.integer('label')
+                    .references('Label.id')
+                    .onDelete('CASCADE');
+            });
+        }
+    });
+
+    knex.schema.hasTable('PurchaseOrder').then(function (exists) {
+        if (!exists) {
+            return knex.schema.createTable('PurchaseOrder', function (t) {
+                t.increments('id').primary();
+                t.text('creator'); // update this when we have user stuff done
```

Comment:
Right now this is just a text field but should probably be some sort of userId

Reply:
Sounds good - maybe add a `TODO:` as part of the comment so we can find it easier later.